pkg/operator/ceph/cluster/cluster.go

```diff
@@ -391,5 +437,11 @@ func (c *cluster) postMonStartupActions() error {
 		return errors.Wrap(err, "failed to enable Ceph messenger version 2")
 	}
 
+	// Remove the root=default and replicated_rule which are created by default. Note that RemoveDefaultCrushMap
+	// ignores some types of errors internally
+	if err := c.removeDefaultCrushMap(); err != nil {
```

Comment:
If there isn't a way to avoid creation of the "default" root, we need to make sure we only remove it if the new setting in the cluster CR is different from the default. For example, most clusters are not going to change the default so we have no reason to remove it. Also note that every time the operator runs a reconcile (e.g. after restart), this code will be triggered and we need to make sure we don't remove the root after OSDs are initialized.

Reply:
Hm, when I ran my tests locally, I found that the ``ceph osd crush rm`` command fails instead of breaking anything if the node is still in use (e.g. because things are nested under it).

Based on that assumption, I think that it’s not a problem to remove that node and the rule always if they’re unused. Was that unclear or do you still think that the removal should be guarded for first_run && crushRoot != default?

Reply:
Yes that sounds good if it's always run and it simply fails silently if it is in use. I believe that just means the error here should be a `logger.Debug()` instead of returning the error.

Reply:
And for the common case where the crush root is just "default", how about if we skip this call?

Reply:
@travisn Hm, I tried to mention in the comment that the removeDefaultCrushMap function will ignore certain errors (in this case, EBUSY and ENOTEMPTY) internally already. The remaining errors are more fatal (e.g. if the ceph CLI can’t reach the mons or something) and those should IMO fail loudly.

If you disagree on that, I’m going to change it though.

Reply:
While I agree that nothing should happen if the default root is being used, I'd rather skip the entire operation if the root is not "default", just so we don't risk anything happening unexpected in the 99% common case.